### PR TITLE
HPCC-28021 fix missing license file in OSX clienttools

### PR DESCRIPTION
**Pull request by xwang2713**
This fix should apply to 8.4.x, 8.2.x, 8.0.x and 7.12.x. I can create each PR if needed.

refactor lib2/CMakeLists.txt which already done for 8.6.x and 8.10.x
Remove no more needed libquadmath, libgfortran and libgcc_s for CBlas lib

Above dependencies are no more required:
hpccdmac01:libs hpccbuild$ otool -L libeclblas.dylib
libeclblas.dylib:
        @rpath/libeclblas.dylib (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libeclrtl.dylib (compatibility version 0.0.0, current version 0.0.0)
        /usr/local/opt/openblas/lib/libopenblas_sandybridgep-r0.2.18.dylib (compatibility version 0.0.0, current version 0.0.0)
        /opt/local/libexec/boost/1.76/lib/libboost_regex-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libnbcd.dylib (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libroxiemem.dylib (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libjlib.dylib (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libyaml.dylib (compatibility version 0.0.0, current version 0.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.50.4)
        /usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib (compatibility version 1.1.0, current version 1.1.0)
        /usr/local/opt/openssl@1.1/lib/libcrypto.1.1.dylib (compatibility version 1.1.0, current version 1.1.0)
        /usr/local/opt/icu4c/lib/libicui18n.63.dylib (compatibility version 63.0.0, current version 63.1.0)
        /usr/local/opt/icu4c/lib/libicuuc.63.dylib (compatibility version 63.0.0, current version 63.1.0)
        /usr/local/opt/icu4c/lib/libicudata.63.1.dylib (compatibility version 63.0.0, current version 63.1.0)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)

Assigned off by xiaoming.wang@lexisnexis.com

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

http://10.224.20.18/job/CE-Clienttools-OSX-8.4.58-1-HPCC-28201/

### CODE REVIEW

**Comment by xwang2713** [comment]
@dcamper I think Michael is in PTO. I know we will have a rc this week. Maybe you can help to review the PR.
The most stuff were already in 8.6.x and later. I just enable them for earlier build and remove unnecessary libs which are not available in the build path and cause the License problem.

**Review by dcamper — APPROVED** [review]
Looks about right.  I have not built everything using these changes as a test, though.

**Review by Michael-Gardner — APPROVED** [review]
@xwang2713 Approved.

File: lib2/CMakeLists.txt [hunk 1]
```diff
@@ -106,6 +106,38 @@ if (WIN32)
   endif()
 endif ()
 
+if (APPLE)
+  set (update_dynamic_shared_lib_path  "execute_process (COMMAND bash \"-c\"  \"otool -L \\\"\${file}\\\" | egrep \\\"(\\\\s|/)\${dylib_name_only}(.[0-9]{1,})*.dylib\\\" | sed \\\"s/^[[:space:]]//g\\\" | cut -d' ' -f1\"
```
Comment (ghalliday) [inline lib2/CMakeLists.txt:110]: Could do with a comment in the cmake file describing what it is doing/why it is needed.

**Comment by ghalliday** [comment]
@xwang2713 should this be applied to 8.6.x and following.
We aren't currently building/supporting 8.4.x - do you know why this was the build that was targetted.  I don't mind merging to 8.4.x (but won't tag a build unless there is a critical reason though.)

**Review by richardkchapman — APPROVED** [review]

**Comment by ghalliday** [comment]
@richardkchapman what is your opinion on this?  All the builds this is targeting we are no longer supporting.  I could apply the change and not tag a new build, but that seems a bit pointless.

**Comment by richardkchapman** [comment]
If only versions affected are no longer supported, and it's client only, seems little point in merging.

**Pull request closed without merging**